### DISCUSSION TITLE
".\CosNotification.idl", line 59: spelling differs from IDL keyword only in case: "EventType"

### DISCUSSION THREAD

**Discussion by xpston008** [DOCGroup/ACE_TAO#2162] [Q&A] [answered]
ace_tao
```
..\bin\tao_idl.exe
TAO_IDL_FE, version 2.5.9 (Based on Sun IDL FE, version 1.3.0)
TAO_IDL_BE, version 2.5.9
```
1、I am a client that needs to access the server
2、In the manufacturer-provided idl file, EventType is defined in omg/CosNotification.idl. 
     The compile command is "... \bin\tao_idl.exe CosNotifyFilter.idl-o.\src\".Error ".\CosNotification.idl", line 59: spelling differs from IDL keyword only in case: "EventType"
    I changed EventType to _EventType in omg/CosNotification.idl, and it compiled normally。As follows:
```
struct EventType
   {
      string domain_name;
      string type_name;
   };

struct _EventType
   {
      string domain_name;
      string type_name;
   };

```
**Question 1：**
     if EventType is changed to _EventType, can the server recognize and transfer data between client and server normally?
**Question 2：**
    idl contains "ace_tao comes with omg idls", e.g #include ".. / orbsvcs orbsvcs/CosNotifyChannelAdmin idl". But I am not sure whether the client accesses the server properly using tao's omg?

Sorry, I'm writing code right now and don't have a debug environment. So I hope I can get an answer. Thanks!
@mitza-oci @jwillemsen

**Comment by iguessthislldo** [comment]
I converted this to a discussion, as there were just questions.

> if EventType is changed to _EventType, can the server recognize and transfer data between client and server normally?

Yes, assuming TAO is handling escaping properly, this is the correct way to use an IDL keyword as an identifier. ~You probably already saw this, but it should be available in C++ as `_cxx_EventType`.~

> idl contains "ace_tao comes with omg idls", e.g #include ".. / orbsvcs orbsvcs/CosNotifyChannelAdmin idl". But I am not sure whether the client accesses the server properly using tao's omg?

No, it shouldn't confuse `_EventType` and `EventType`, if that's what you're asking.

Adam or Johnny might have some more input if I'm missing something about how this works specifically with TAO. We have had some issues with supporting escaped IDL keywords in the past in OpenDDS, but I'm not sure about TAO.

**Reply by xpston008** [reply]
Thank you!
1 and 2 are two different questions! There is no correlation!
In response to the first question, I don't understand what you mean!
Servant Developed by another manufacturer. The name "EventType" is used for servants. The client (which I developed) uses the name "_EventType". Whether the data transfer can be interpreted correctly when the client variable names are different from the server variable names.

**Reply by iguessthislldo** [reply]
Apologies I misread what you were asking. I thought you were asking if renaming it was the right thing to do. Also ignore `_cxx_EventType`. I was thinking of using C++ keywords, like `class` in IDL.

Rereading you first question, for the data transfer, the CDR encoding of both structs should be the exact same. Names of types and members shouldn't affect that. Also looking through the generated code from `tao_idl` with a file with a `struct EventType {...};` I don't see anything where `_EventType` would be different from `EventType` if it wasn't a keyword. It never mentions `_EventType`, only `EventType`, so the ORB shouldn't know the difference.

I'm still not sure what you mean in question 2 though. Maybe @jwillemsen or @mitza-oci would know what you're talking about.

**Reply by mitza-oci** [reply]
Every time you use CORBA (TAO or otherwise) you can decide to use the exact same IDL file or an equivalent IDL file.

**Reply by xpston008** [reply]
@mitza-oci  @iguessthislldo 
Thanks for the answer！！


_**".\CosNotification.idl", line 59: spelling differs from IDL keyword only in case: "EventType"**_

**idls are provided by other manufacturers(Server)**.How do I handle this error when compiling idl on **my client**?

After the following modification,idls compiles without errors, but the data communication between the client and the server is normal?
```
//EventType is changed to _cxx_EventType  
struct EventType // old
{
string domain_name;
string type_name;
};

struct _cxx_EventType //new
{
string domain_name;
string type_name;
};

```

``

**Reply by mitza-oci** [reply, marked as answer]
Because of the fact that we previously discussed...
> Every time you use CORBA (TAO or otherwise) you can decide to use the exact same IDL file or an equivalent IDL file.
...you can change the IDL file to avoid that error

Also, see the command line options for `tao_idl` which are documented in `$TAO_ROOT/docs/compiler.html`.

**Reply by xpston008** [reply]
Thanks for the answer！！
@mitza-oci  thanks!  I think my problem has been solved！

add -Cw  Option 

Description
<html>
<body>

 Output a warning if two identifiers in the same scope differ in spelling only by case (default is output of error message).
--

</body>
</html>
